### DISCUSSION TITLE
What is DivestOS's encryption strength compared to veracrypt, or GrapheneOS? or a device with Titan M2?

### DISCUSSION THREAD

**Discussion by Username23453** [Divested-Mobile/DivestOS-Build#183] [General]
And how many characters in a password can you choose for DivestOS?

**Comment by SkewedZeppelin** [comment]
https://source.android.com/docs/security/features/encryption

Whether FBE or FDE is used depends on device.
Key storage and whether it is software or hardware backed depends on device.

FBE has more padding, from GrapheneOS, on 16.0+: https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Patches/LineageOS-20.0/android_system_extras/0001-ext4_pad_filenames.patch

Max passwords length is 64 characters, increased from 16, from GrapheneOS.

**Reply by Username23453** [reply]
@SkewedZeppelin,Thanks! What do u mean by "more padding"?
2.) I'd want to be able to use the camera without having to login, but also have the pics encrypted instantly & sent to a certain login/profile & I'd like to be able to use a streaming audio app while logged out too. And I'd like to be able to have a decoy & a hidden login/profile, would FBE or FDE be more compatible with these needs? And is DivestOS compatible with all this?  Sorry for the questions, I appreciate your work, I'm new to Divest, just recently heard of it. I **am** writing down/taking notes on all this.

**Reply by SkewedZeppelin** [reply]
None of that is feasible.

**Reply by Username23453** [reply]
@SkewedZeppelin, Thanks, What do u mean by "more padding"?
